tfx_addons/predictions_to_bigquery/utils.py

```diff
@@ -184,64 +174,90 @@ def convert_tensorflow_dtype_to_bq_type(tf_dtype: tf.dtypes.DType) -> str:
-    """
+def _feature_spec_to_bq_schema_fields(feature_spec: FeatureSpec,
+                                      required: bool = True) -> List[Dict]:
+  """Convert a TensorFlow feature spec to a list of BigQuery schema fields.
+
+  Args:
+    feature_spec: TensorFlow feature spec.
+    required: Whether the field is required.
+
+  Returns:
+      A list of BigQuery schema fields.
+  """
   return [{
-      "name": feature_name,
-      "type": convert_tensorflow_dtype_to_bq_type(feature_def.dtype),
+      "name": get_bq_field_name_from_key(feature_name),
+      "type": _convert_tensorflow_dtype_to_bq_type(feature_def.dtype),
       "mode": "REQUIRED" if required else "NULLABLE",
-  } for feature_name, feature_def in features.items()]
+  } for feature_name, feature_def in feature_spec.items()]
 
 
-def create_annotation_fields(
-    label_field_name: str = "category_label",
-    score_field_name: str = "score",
+def _create_annotation_fields(
+    *,
     required: bool = True,
+    add_label_field: bool = False,
     add_datetime_field: bool = True,
 ) -> List[Dict]:
-  """
-    Create a list of BigQuery schema fields for the annotation fields.
+  """Creates a list of annotation fields in BigQuery schema formatkjjjj.
 
-    Args:
-        label_field_name: The name of the label field.
-        score_field_name: The name of the score field.
-        required: Whether the fields are required.
-        add_datetime_field: Whether to add a datetime field.
+  Args:
+      label_field_name: The name of the label field.
+      score_field_name: The name of the score field.
+      required: Whether the fields are required.
+      add_datetime_field: Whether to add a datetime field.
 
-    Returns:
-        A list of BigQuery schema fields.
-    """
+  Returns:
+      A list of BigQuery schema fields.
+  """
 
-  label_field = {
-      "name": label_field_name,
-      "type": "STRING",
-      "mode": "REQUIRED" if required else "NULLABLE",
-  }
+  fields = []
+  if add_label_field:
+    label_field = {
+        'name': 'category_label',
+        'type': 'STRING',
+        'mode': 'REQUIRED' if required else 'NULLABLE',
+    }
+    fields.append(label_field)
 
   score_field = {
-      "name": score_field_name,
-      "type": "FLOAT",
-      "mode": "REQUIRED" if required else "NULLABLE",
+      'name': 'score',
+      'type': 'FLOAT',
+      'mode': 'REQUIRED' if required else 'NULLABLE',
   }
-
-  fields = [label_field, score_field]
+  fields.append(score_field)
 
   if add_datetime_field:
     datetime_field = {
-        "name": "datetime",
-        "type": "TIMESTAMP",
-        "mode": "REQUIRED" if required else "NULLABLE",
+        'name': 'datetime',
+        'type': 'TIMESTAMP',
+        'mode': 'REQUIRED' if required else 'NULLABLE',
     }
     fields.append(datetime_field)
 
   return fields
+
+
+def feature_spec_to_bq_schema(feature_spec: FeatureSpec,
+                              required: bool = False,
```

Comment:
I don't think `required` is used anywhere in the component where this util is called. I may be mistaken.